enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,434 @@
+
+### Non-Goals
+
+**Baremetal bootstrap and control plane nodes.**
+
+This proposal focuses on the OpenShift compute nodes only. Deploying or
+migrating the Control plane to BMs is out of scope.
+
+**Baremetal deployments on anything other than an OpenStack tenant**
+
+There are other efforts that provide OpenShift on the bare metal. See
+[Metal³][metal3-io] and the [Baremetal installers](#baremetal-ipi--upi).
+
+These projects allow deploying OpenShift in an environment where all the nodes
+are physical and there is no OpenStack or other cloud layer in-between.
+
+In contrast, this proposal uses the existing OpenStack provider and extends it
+to allow adding baremetal workers if the cloud supports it.
+
+**Attaching floating IPs to the BM workers.**
+
+The OpenStack cluster might make the baremetal nodes accessible to the outside
+world, but exposing them via floating IPs might represent an increase in
+complexity and is currently not an explicit goal.
+
+## Proposal
+
+This enhancement is less about adding a new logical piece, than it is about
+uncovering unknowns in a configuration that might work out-of-the-box in
+selected environments.
+
+Therefore, the implementation consists of:
+* identifying a reference architecture
+* anticipating probable issues
+* testing and fixing errors
+
+### Characteristics of the reference architecture
+
+Prerequisites:
+
+* The OpenStack cloud runs the [Bare Metal (ironic)][openstack-ironic] service
+  to provide baremetal management
+* The OpenStack [Compute (nova)][openstack-nova] and [Networking
+  (neutron)][openstack-neutron] services are configured to serve bare metal
+  nodes using the Nova API
+* All of the additional complexity inherent to baremetal is handled by the IaaS
+  cloud and its administrators:
+    * Enrolling new hardware and discovering its components
+    * Wiping the disks between deployments
+    * Booting the system up from the desired image
+    * Networking
+* The choice between deploying a virtual or baremetal machine is done by
+  selecting an appropriate Nova flavor and Neutron network
+* Both virtual and baremetal machines consume the same image from glance
+* The installer will have no insight into the Ironic networking
+    * All it will do is create a node with the user-specified flavor and
+      network
+
+We target four distinct use-cases:
+
+1. Install with BM workers on the installer-provisioned network
+2. Install with BM workers on a preexisting network
+3. Install with both VM and BM workers (BMs on a preexisting network)
+4. Install on VMs then add BM workers (BMs on a preexisting network)
+
+### Implementation work
+
+#### Case 1. Install with BM workers on the installer-provisioned network
+
+Booting BMs in a virtual network requires specific hardware; this case is a
+simplified one and is not expected to be the most represented in real-world
+installations.
+
+Prerequisites for the user:
+
+* Being able to provision BM nodes on a tenant network
```

Comment:
We may want to be more explicit and say that Ironic needs to be able to listen and PXE-boot nodes in freshly created networks (created by the installer).
@juliakreger would that work out of the box when the hardware supports it or does the installer need to do anything special?